.editorconfig

```diff
@@ -12,3 +12,8 @@ insert_final_newline = true
 [*.php]
 indent_style = space
 indent_size = 4
+
+# 2 space indentation
```

Comment:
This applies to all code-blocks in markdown files or is there a way to tell editorconfig to lookup the corresponding type instead? For example: php code blocks -> look at *.php config?

Reply:
It applies to the whole markdown document and you cant have mixed values. So you need to either default to 2 or 4.

2 = You will need to double indent php and other types of code blocks. Also affects the number of indents needed to create "git_hook-variables" in the doc/parameters.

4 = You need to press space two times for all yaml code blocks and tab (4 spaces) for the rest.

Currently there are more 2 space indentation than 4, but there are no perfect solution. :(

Reply:
I'm a bit confused by this explanation to be honest.

Would it make sense to set the defaults to always be 4 spaces as indentation?

That way it would be compatible with how php is written (psr-12) and with the defaults of the symfony yaml dumper: https://github.com/symfony/yaml/blob/6.2/Dumper.php#L30
JSON would also be indented by 4 spaces.

That way, we might be able to automatically run an editor config checker on this project?

Reply:
Sorry about that! If you want to automatically correct it using .editorconfig it needs to be set to 4 or PHP-code blocks won't match PSR-12. The official specification of YAML specify `zero or more` for their [indentation](https://yaml.org/spec/1.2.2/#61-indentation-spaces). Both `2` or `4` are correct. Most .yml files I have come across uses 2 spaces, like the one used in this project.  

The debate on how many spaces you should indent your code with are "tab vs spaces" once again. `package.json` are a valid JSON and that one are 2 spaces by default.

Reply:
OK, thanks for clarifying.

I'dd say lets go for 4 spaces then to conclude this debate.

Reasoning:

* .php 4 by PSR-12
* .yaml 4 is the symfony convention
* composer.json -> 4 by default

Sounds like the most logical and straight forward choice.